package/src/services/check-parser/parser.ts

```diff
@@ -46,89 +56,110 @@ export class Parser {
+      const item = bfsQueue.shift()!
+      const { localDependencies, npmDependencies, error } = Parser.parseDependencies(item.filePath, item.content)
+      if (error) {
+        collector.addParsingError(item.filePath, error.message)
+        continue
+      }
+      const unsupportedDependencies = npmDependencies.filter((dep) => !this.supportedModules.has(dep))
+      if (unsupportedDependencies.length) {
+        collector.addUnsupportedNpmDependencies(item.filePath, unsupportedDependencies)
+      }
+      const localDependenciesResolvedPaths: Array<{filePath: string, content: string}> = []
+      localDependencies.forEach((localDependency: string) => {
+        const filePath = path.join(path.dirname(item.filePath), localDependency)
+        try {
+          const dep = Parser.readDependency(filePath, extension)
+          localDependenciesResolvedPaths.push(dep)
+        } catch (err: any) {
+          collector.addMissingFile(filePath)
         }
-        const localDependenciesResolvedPaths = localDependencies.map((localDependency: string) => {
-        // Convert a relative path to an absolute path.
-          const resolvedPath = path.join(path.dirname(currentPath), localDependency)
-          // The import call may not have the .js / .ts extension, so we add it here.
-          // We assume that the other file is also a JS or typescript file based on the entrypoint file type.
-          // At some point, though, it may make sense to add support for mixing JS and typescript.
-          return Parser.addExtension(extension, resolvedPath)
-        })
-        localDependenciesResolvedPaths.forEach((path: string) => {
-          if (!collector.hasDependency(path)) {
-            collector.addDependency(path)
-            bfsQueue.push(path)
-          }
-        })
-      } catch (err: any) {
-        if (err.code === 'ENOENT') {
-          collector.addMissingFile(currentPath)
-        } else {
-          collector.addParsingError(currentPath, err.message)
+      })
+      localDependenciesResolvedPaths.forEach(({ filePath, content }: {filePath: string, content: string}) => {
+        if (collector.hasDependency(filePath)) {
+          return
         }
-      }
+        collector.addDependency(filePath, content)
+        bfsQueue.push({ filePath, content })
+      })
     }
 
     collector.validate()
 
     return collector.getDependencies()
   }
 
-  static addExtension (extension: string, filePath: string) {
-    if (!filePath.endsWith(extension)) {
-      return filePath + extension
+  static readDependency (filePath: string, preferedExtenstion: SupportedExtensions) {
+    // Read the specific file if it has an extension
+    if (path.extname(filePath).length) {
+      const content = fs.readFileSync(filePath, { encoding: 'utf-8' })
+      return { filePath, content }
     } else {
-      return filePath
+      if (preferedExtenstion === SupportedExtensions.JS) {
```

Comment:
Could it ever be the case that the main entrypoint check is in JS, but it imports TS files? Or is it only possible for TS to import JS?

Reply:
That shouldn't be allowed. We will experience probably different resolving issues in the future but js import ts in a pure matter shouldn't be possible